independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/StereotypeRegistrar.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.arc.processor;
+
+import java.util.Set;
+import javax.enterprise.inject.Stereotype;
+import org.jboss.jandex.DotName;
+
+/**
+ * Makes it possible to turn an annotation into a stereotype without adding a {@link Stereotype} annotation to it.
+ */
+public interface StereotypeRegistrar extends BuildExtension {
+
+    /**
+     * Returns a set of annotation types (their names) that should be treated as stereotypes.
+     * To modify (meta-)annotations on these annotations, use {@link AnnotationsTransformer}.
```

Comment:
This is pretty smart (and I can see that it's even tested ;-)

Reply:
This is a very nice thing about ArC's extensions: annotation transformers apply to annotations. Portable Extensions don't allow that, and so to allow portability, neither do Build Compatible Extensions :-/